test/unit/SSHConnection.test.ts

```diff
@@ -0,0 +1,7 @@
+describe('SSHConnection', () => {
```

Comment:
Low on time. There's no other functionality needed to implement them but I just wanted to get the code out quickly.

Reply:
Also, this made me realize that they should really be integration tests, not unit. I split out the pure-logic part of it, wrote unit tests for that, and moved this into `test/integ` instead of `test/unit`.